benchmarks/gems_master_with_spatial_parallelism/benchmark_amoebanet_gems_master_with_sp.py

```diff
@@ -0,0 +1,485 @@
+# Copyright 2023, The Ohio State University. All rights reserved.
+# The MPI4DL software package is developed by the team members of
+# The Ohio State University's Network-Based Computing Laboratory (NBCL),
+# headed by Professor Dhabaleswar K. (DK) Panda.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.distributed as dist
+import torchvision.transforms as transforms
+import torchvision
+import numpy as np
+import time
+import sys
+import math
+import logging
+from torchgems import parser
+from torchgems.mp_pipeline import model_generator
+from torchgems.train_spatial import get_shapes_spatial, split_input
+from torchgems.train_spatial_master import (
+    train_spatial_model_master,
+    verify_spatial_master_config,
+)
+import torchgems.comm as gems_comm
+
+parser_obj = parser.get_parser()
+args = parser_obj.parse_args()
+
+if args.verbose:
+    logging.basicConfig(level=logging.DEBUG)
+
+if args.halo_d2:
+    from models import amoebanet
+    from models import amoebanet_d2
+
+else:
+    from models import amoebanet
+
+gems_comm.initialize_cuda()
+
+
+class Unbuffered(object):
+    def __init__(self, stream):
+        self.stream = stream
+
+    def write(self, data):
+        self.stream.write(data)
+        self.stream.flush()
+
+    def writelines(self, datas):
+        self.stream.writelines(datas)
+        self.stream.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stream, attr)
+
+
+def init_processes(backend="tcp"):
```

Comment:
this default should always be `mpi`. Change https://github.com/OSU-Nowlab/MPI4DL/blob/c6716708b11501fe6a6b995b39152d6d0e636176/benchmarks/spatial_parallelism/benchmark_amoebanet_sp.py#L65 as well